vertx-sql-client-template/src/main/java/io/vertx/sqlclient/template/generator/RowMapperGen.java

```diff
@@ -0,0 +1,215 @@
+              bilto4(writer, meth, prop, "java.util.Arrays.stream((" + rowType + "[])val).map(elt -> " + wrapExpr(prop.getType(), "elt") + ").collect(java.util.stream.Collectors.toCollection(java.util.ArrayList::new))");
+            }
+            break;
+          }
+          case SET: {
+            String meth = getArrayType(prop.getType());
+            if (meth != null) {
+              bilto4(writer, meth, prop, "java.util.Arrays.stream((" + rowType + "[])val).map(elt -> " + wrapExpr(prop.getType(), "elt") + ").collect(java.util.stream.Collectors.toCollection(java.util.HashSet::new))");
+            }
+            break;
+          }
+        }
+      });
+    model
+      .getPropertyMap()
+      .values()
+      .stream()
+      .filter(prop -> PK.contains(prop.getKind()))
+      .filter(prop -> prop.isAdder() && !prop.isSetter())
+      .forEach(prop -> {
+        String meth = getArrayType(prop.getType());
+        String rowType = rowType(prop.getType());
+        if (meth != null) {
+          String columnName = getMappingName(prop, Column.class.getName());
+          if (columnName != null) {
+            writer.print("    val = row." + meth + "(\"" + columnName + "\");\n");
+            writer.print("    if (val != null) {\n");
+            writer.print("      for (" + rowType + " elt : (" + rowType + "[])val) {\n");
+            writer.print("        obj." + prop.getAdderMethod() + "(" + wrapExpr(prop.getType(), "elt") + ");\n");
+            writer.print("      }\n");
+            writer.print("    }\n");
+          }
+        }
+      });
+  }
+
+  private void bilto4(PrintWriter writer, String meth, PropertyInfo prop, String converter) {
+    String columnName = getMappingName(prop, Column.class.getName());
+    if (columnName != null) {
+      writer.print("    val = row." + meth + "(\"" + columnName + "\");\n");
+      writer.print("    if (val != null) {\n");
+      writer.print("      obj." + prop.getSetterMethod() + "(" + converter +  ");\n");
+      writer.print("    }\n");
+    }
+  }
+
+  private static String wrapExpr(TypeInfo type, String expr) {
+    DataObjectInfo dataObject = type.getDataObject();
+    if (dataObject != null) {
+      MapperInfo deserializer = dataObject.getDeserializer();
+      if (deserializer != null) {
+        if (deserializer.getKind() == MapperKind.SELF) {
+          return "new " + type.getName() + "(" + expr + ")";
+        } else {
+          return deserializer.getQualifiedName() + "." + String.join(".", deserializer
+            .getSelectors()) + "(" + expr + ")";
+        }
+      }
+      throw new UnsupportedOperationException();
+    } else {
+      return expr;
+    }
+  }
+
+  private static String rowType(TypeInfo type) {
+    DataObjectInfo dataObject = type.getDataObject();
+    if (dataObject != null) {
+      return rowType(dataObject.getJsonType());
+    }
+    return type.getName();
+  }
+
+  private static String getter(TypeInfo type) {
+    switch (type.getKind()) {
```

Comment:
it seems we don't support mapping string/numeric datat type to enum type(name/ordinal). I wonder if we could support this here.

Reply:
I think we can map it to get/set String easily

Reply:
at least for beginning, some might want to provide at some point a custom function for this type

Reply:
I've checked in such mapping support in master for PostgreSQL and implement support for templates in this PR.